org.lflang.tests/src/org/lflang/tests/LFParsingTest.java

```diff
@@ -13,19 +13,108 @@
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+
 import org.lflang.lf.Model;
 
+import com.google.inject.Inject;
+
 @ExtendWith(InjectionExtension.class)
 @InjectWith(LFInjectorProvider.class)
 public class LFParsingTest {
-	@Inject
-	private ParseHelper<Model> parseHelper;
-	
-	@Test
-	public void loadModel() throws Exception {
-		Model result = parseHelper.parse("Hello Xtext!");
-		Assertions.assertNotNull(result);
-		List<Diagnostic> errors = result.eResource().getErrors();
-		Assertions.assertTrue(errors.isEmpty(), "Unexpected errors: " + IterableExtensions.join(errors, ", "));
-	}
+
+    @Inject
+    private ParseHelper<Model> parseHelper;
+
+
+    @Test
+    public void testLexingLifetimeAnnots() throws Exception {
+        assertNoParsingErrorsIn(makeLfTargetCode("Rust",
+                                                 "        struct Hello<'a> { \n"
+                                                     + "            r: &'a str,\n"
+                                                     + "            r2: &'a Box<Hello<'a>>,\n"
+                                                     + "        }"));
+    }
+
+
+    @Test
+    public void testLexingSingleLifetimeAnnot() throws Exception {
+        // just to be sure, have a single lifetime annot.
+        assertNoParsingErrorsIn(makeLfTargetCode("Rust",
+                                                 "        struct Hello { \n"
+                                                     + "            r: &'static str,\n"
+                                                     + "        }"));
+    }
+
+
+    @Test
+    public void testLexingNewlineCont() throws Exception {
+        /*
+        This example looks like this:
+        "a\
+        bcde"
+
+        This is valid C++ to escape a newline.
+         */
+
+        assertNoParsingErrorsIn(makeLfTargetCode("Cpp",
+                                                 "        \"a\\\n"
+                                                     + "        bcde\"\n"
+        ));
+    }
+
+
+    @Test
+    public void testLexingSquotedString() throws Exception {
+        // we can't do that anymore
+        expectParsingErrorIn(makeLfTargetCode("Python", "a = ' a string '"));
```

Comment:
I think not supporting `'` and having to use `"` is not ideal, but something we can live with. But not supporting `"'...'"` seems bad to me. Sometime this is simply needed... . Would `"\'...\'"` work?

Reply:
it is supported, see your other comment ^^ the `"` you're seeing are in Java, not LF